signer/src/api.rs

```diff
@@ -117,67 +80,32 @@ impl MessageParams {
     }
 }
 
-/// Unsigned message api structure
-#[cfg_attr(feature = "with-arbitrary", derive(arbitrary::Arbitrary))]
-#[derive(Debug, Clone, Deserialize, PartialEq, Serialize)]
-pub struct UnsignedMessageAPI {
-    #[serde(alias = "To")]
-    pub to: String,
-    #[serde(alias = "From")]
-    pub from: String,
-    #[serde(alias = "Nonce")]
-    pub nonce: u64,
-    #[serde(alias = "Value")]
-    pub value: String,
-
-    #[serde(rename = "gaslimit")]
-    #[serde(alias = "gasLimit")]
-    #[serde(alias = "gas_limit")]
-    #[serde(alias = "GasLimit")]
-    pub gas_limit: i64,
-
-    #[serde(rename = "gasfeecap")]
-    #[serde(alias = "gasFeeCap")]
-    #[serde(alias = "gas_fee_cap")]
-    #[serde(alias = "GasFeeCap")]
-    pub gas_fee_cap: String,
-
-    #[serde(rename = "gaspremium")]
-    #[serde(alias = "gasPremium")]
-    #[serde(alias = "gas_premium")]
-    #[serde(alias = "GasPremium")]
-    pub gas_premium: String,
-
-    #[serde(alias = "Method")]
-    pub method: u64,
-    #[serde(alias = "Params")]
-    pub params: String,
+/// Structure containing an `UnsignedMessageAPI` or a `SignedMessageAPI`
+#[derive(Deserialize, Serialize)]
+#[serde(untagged)]
+pub enum MessageTxAPI {
+    #[serde(with = "MessageAPI")]
+    Message(Message),
+    #[serde(with = "SignedMessageAPI")]
+    SignedMessage(SignedMessage),
 }
 
-/// Signature api structure
-#[cfg_attr(feature = "with-arbitrary", derive(arbitrary::Arbitrary))]
-#[derive(Debug, Clone, Deserialize, PartialEq, Serialize)]
-pub struct SignatureAPI {
-    #[serde(rename = "type")]
-    pub sig_type: u8,
-    #[serde(with = "serde_base64_vector")]
-    pub data: Vec<u8>,
+#[derive(Deserialize, Serialize)]
+#[serde(untagged)]
+pub enum MessageTx {
+    Message(Message),
+    SignedMessage(SignedMessage),
 }
 
-/// Signed message api structure
-#[cfg_attr(feature = "with-arbitrary", derive(arbitrary::Arbitrary))]
-#[derive(Debug, Clone, Deserialize, PartialEq, Serialize)]
-pub struct SignedMessageAPI {
-    pub message: UnsignedMessageAPI,
-    pub signature: SignatureAPI,
-}
+impl TryFrom<MessageTx> for MessageTxAPI {
```

Comment:
Is this even fallible? It doesn't look like it is so why not `impl From<MessageTx> for MessageTxAPI` ?